src/EditorFeatures/Core.Wpf/Adornments/AdornmentManager.cs

```diff
@@ -22,42 +22,26 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Adornments
     /// <summary>
     /// UI manager for graphic overlay tags. These tags will simply paint something related to the text.
     /// </summary>
-    internal class AdornmentManager<T> where T : GraphicsTag
+    internal abstract class AdornmentManager<T> where T : GraphicsTag
     {
-        private readonly object _invalidatedSpansLock = new object();
+        protected readonly object _invalidatedSpansLock = new object();
 
-        private readonly IThreadingContext _threadingContext;
+        protected IThreadingContext ThreadingContext { get; }
 
         /// <summary>View that created us.</summary>
-        private readonly IWpfTextView _textView;
+        protected IWpfTextView TextView { get; }
 
         /// <summary>Layer where we draw adornments.</summary>
-        private readonly IAdornmentLayer _adornmentLayer;
+        protected IAdornmentLayer AdornmentLayer { get; }
 
         /// <summary>Aggregator that tells us where to draw.</summary>
-        private readonly ITagAggregator<T> _tagAggregator;
+        protected ITagAggregator<T> TagAggregator { get; }
 
         /// <summary>Notification system about operations we do</summary>
-        private readonly IAsynchronousOperationListener _asyncListener;
+        protected IAsynchronousOperationListener AsyncListener { get; }
```

Comment:
are all of these needed by the subclasses?

Reply:
I made the items that are not used by the subclass private readonly again.